bench-tps/src/bench.rs

```diff
@@ -386,6 +396,38 @@ fn generate_txs(
     }
 }
 
+fn poll_blockhash<T: Client>(
+    exit_signal: &Arc<AtomicBool>,
+    blockhash: &Arc<RwLock<Hash>>,
+    client: &Arc<T>,
+    id: &Pubkey,
+) {
+    let mut blockhash_time;
+    loop {
+        blockhash_time = Instant::now();
+        loop {
+            let old_blockhash = *blockhash.read().unwrap();
+            if let Ok((new_blockhash, _fee)) = client.get_new_blockhash(&old_blockhash) {
+                *blockhash.write().unwrap() = new_blockhash;
+                break;
+            } else {
+                if blockhash_time.elapsed().as_secs() > 30 {
+                    panic!("Blockhash is not updating");
+                }
+                sleep(Duration::from_millis(100));
```

Comment:
Maybe, use 50ms sleep here. The slot duration in v21 was ~250ms. Now we are seeing between 400/500ms. The 50ms granularity would cover more variations in slot time.

Reply:
Done